Goal: Always put the "ro|rw" option in mtab

Fixes: ?

Status wrt upstream: Should be forwarded if pertinent

Note: 

Index: samba-3.0.25rc2/source/client/smbmnt.c
===================================================================
--- samba-3.0.25rc2.orig/source/client/smbmnt.c	2007-04-24 11:07:00.470062591 +0200
+++ samba-3.0.25rc2/source/client/smbmnt.c	2007-04-24 11:07:52.143973086 +0200
@@ -276,7 +276,10 @@
         ment.mnt_fsname = share_name ? share_name : "none";
         ment.mnt_dir = mount_point;
         ment.mnt_type = "smbfs";
-        ment.mnt_opts = "";
+	if (mount_ro)
+                ment.mnt_opts = "ro";
+        else
+                ment.mnt_opts = "rw";
         ment.mnt_freq = 0;
         ment.mnt_passno= 0;
 
